### PR TITLE
Update CLI Build

### PR DESCRIPTION
**Pull request by dubeejw**
- Place binary at the root of uncompressed folder
- Rename compressed files
- Make tar.gz extensions .tgz
- Compress Mac binary with zip

Fixes https://github.com/openwhisk/openwhisk/issues/1039

### CODE REVIEW

File: tools/go-cli/Dockerfile [hunk 1]
```diff
@@ -32,38 +32,59 @@ RUN cd /src/github.com/go-whisk-cli && echo "Dependencies list requires restruct
...
-      tar -cf build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar build/$GOOS/$GOARCH/$EXEC && \
-      gzip < build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar > build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar.gz && \
-      rm -f build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar; \
+    compress_binaries() { \
+        comp_name=$(get_compressed_name $1 $2) && \
+        cd build/$1/$2 && \
+        if [ $4 = tgz ]; then \
+            echo "Compressing binary '$3' to '$comp_name.tgz'" && \
+            tar -cf $comp_name.tar $3 && \
+            gzip < $comp_name.tar > $comp_name.tgz && \
```
Comment (markusthoemmes) [inline tools/go-cli/Dockerfile]: Why not `tar -cvzf $comp_name.tgz $3` instead of two commands?

**Comment by markusthoemmes** [comment]
Would it make sense to externalize all the scripting into a bash script to be able to omit all the `\` and `&&` for improved readability?

**Comment by dubeejw** [comment]
@mt, do you know how to echo information from an externalized .sh file so that it will show in the terminal output in real-time?

File: tools/go-cli/Dockerfile [hunk 2]
```diff
@@ -32,38 +32,58 @@ RUN cd /src/github.com/go-whisk-cli && echo "Dependencies list requires restruct
...
-      go build -ldflags "-X main.CLI_BUILD_TIME=`date -u '+%Y-%m-%dT%H:%M:%S%:z'`" -v -o build/$GOOS/$GOARCH/$EXEC main.go && \
-      tar -cf build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar build/$GOOS/$GOARCH/$EXEC && \
-      gzip < build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar > build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar.gz && \
-      rm -f build/$GOOS/$GOARCH/openwhisk-$GOOS-$GOARCH.tar; \
+    compress_binaries() { \
+        comp_name=$(get_compressed_name $1 $2) && \
+        cd build/$1/$2 && \
+        if [ $4 = tgz ]; then \
+            echo "Compressing binary '$3' to '$comp_name.tgz'" && \
+            tar -cvzf $comp_name.tgz $3 && \
```
Comment (csantanapr) [inline tools/go-cli/Dockerfile]: Creating this tgz with this command will give us the auto expand like in Safari Downloads?

**Comment by csantanapr** [comment]
@dubeejw try putting `set +x` at the top of the shell script to see if it shows the output that you are looking for

**Comment by csantanapr** [comment]
@dubeejw I agree with @mt it's better to put the build code into a script and RUN from Dockerfile.

I was able to get it working and was able to get all the output from the command, and you can see all the progress from the go build step

Take a look at my branch https://github.com/openwhisk/openwhisk/compare/master...csantanapr:gocli_build_script

**Comment by dubeejw** [comment]
@csantanapr, thanks. I'll try it out.

**Comment by dubeejw** [comment]
PG1 494.

**Comment by dubeejw** [comment]
Anything else on this one?

**Comment by csantanapr** [comment]
rebase and merged 06225e2b51944f73e67ed38e93249c129643456b

**Pull request closed without merging**